### PR TITLE
Displaying the Battle Screen - US1

### LINKED ISSUE #2: Displaying the Battle Screen
## Value Proposition

- As a player
- I want to see the battle screen
- in order to prepare for the upcoming battle.

## Description

us1

## Acceptance criteria

- [x] The battle screen should display the player's and opponent's 
        Pokemon, their names, and their HP bars.

## Tasks

- [x] Create the battle screen component.
- [x] Implement logic to display the Pokemon, their names, and their 
        HP bars.
- [x] Write tests to ensure the functionality is working as intended.
- [x] Create a  folder with ressources.

### PR DESCRIPTION
**Pull request by dennisbuchwald**
Create the battlescreen without function for my capestone project.
At the moment the CSS is not final and the overlay is missing. 

The tests are still missing, but I will write them as soon as the function is there, because at the moment I am not sure if all elements can stay as they are, as soon as the functionality is added. 

Here is the US #2 
[here](https://capstone-project-pokemon-battler-app-git-6cb244-dennisbuchwald.vercel.app) you can test the app.

edit after first review:
- I've redesigned the app from the ground up.
- All css.modules have been removed and replaced with style components.
- In addition, I have freed the HTML (almost completely) from <div> and replaced it with semantic HTML.
- The healthbar was reworked and got a useState, so I can manipulate it later and change the display.

### CODE REVIEW

**Comment by dennisbuchwald** [comment]
Okay I just see at Vercel my Google Font is not displayed, too bad but not so important since I will probably finally use another one anyway. 

But that comes only when I completely revise the CSS. :-)

File: components/BattleScreen/EnemyState/enemystate.module.css [hunk 1]
```diff
@@ -0,0 +1,96 @@
+@import url("https://fonts.googleapis.com/css2?family=Press+Start+2P&display=swap");
```
Comment (PReichetanz) [inline components/BattleScreen/EnemyState/enemystate.module.css]: This will be a problem because you don't ask the user for permission to request a Google server. You can avoid this by using `next/font` instead.

File: components/BattleScreen/EnemyState/enemystate.module.css [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+.enemy--state-box {
```
Comment (PReichetanz) [inline components/BattleScreen/EnemyState/enemystate.module.css]: Please note that one requirement for the app is using styled-components instead of BEM.

File: components/BattleScreen/EnemyState/enemystate.module.css [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+.enemy--stats-wrapper {
+	position: relative;
+}
+
+/* .enemy--stats-wrapper:after {
+	content: "";
+	clear: both;
+	display: table;
+} */
```
Comment (PReichetanz) [inline components/BattleScreen/EnemyState/enemystate.module.css]: Please remove unused code.

File: components/BattleScreen/EnemyState/EnemyState.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import styles from "./enemystate.module.css";
+
+export default function EnemyState() {
+	return (
+		<div className={styles["enemy--state-box"]}>
```
Comment (PReichetanz) [inline components/BattleScreen/EnemyState/EnemyState.js]: Wouldn't be a `section` or an `article` better as a wrapper?

File: components/BattleScreen/EnemyState/EnemyState.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+				<p
+					class="enemy--pokemon-health-bar"
+					role="progressbar"
+					aria-valuenow="80"
+					aria-valuemin="0"
+					aria-valuemax="80"
+				></p>
+				<p className={styles["enemy--pokemon-slash-health"]}>/</p>
+				<p class={styles["enemy--pokemon-max-health"]}>120</p>
+			</div>
```
Comment (PReichetanz) [inline components/BattleScreen/EnemyState/EnemyState.js]: This rather looks like a list ;)

File: components/BattleScreen/Menu/Menu.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+				<div className={styles["menu--overview-box-left"]}>
+					{/* <p>Klicke auf Kampf um anzugreifen</p> */}
+				</div>
+				<div className={styles["menu--overview-box-right"]}>
+					{/* <div className={`${styles.btn} ${styles["btn--fight"]}`}>Kampf</div>
+					<div className={`${styles.btn} ${styles["btn--bag"]}`}>Beutel</div>
+					<div className={`${styles.btn} ${styles["btn--pokemon"]}`}>
+						POKéMON
+					</div>
+					<div className={`${styles.btn} ${styles["btn--run"]}`}>Flucht</div> */}
```
Comment (PReichetanz) [inline components/BattleScreen/Menu/Menu.js]: Please remove unused code.

File: components/BattleScreen/Menu/menu.module.css [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+	opacity: 0.8;
+	cursor: pointer;
+	text-transform: uppercase; /* um alle Texte in Großbuchstaben umzuwandeln */
+}
+
+.btn--fight {
+	color: black;
+	left: 0px;
+	top: 5px;
+	text-transform: uppercase;
```
Comment (PReichetanz) [inline components/BattleScreen/Menu/menu.module.css:71]: This should not be necessary if the `.btn` class already has this rule.

File: components/BattleScreen/PlayerState/PlayerState.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import styles from "./playerstate.module.css";
+
+export default function PlayerState() {
+	return (
+		<div className={styles["player--state-box"]}>
+			<div className={styles["player--state-wrapper"]}>
```
Comment (PReichetanz) [inline components/BattleScreen/PlayerState/PlayerState.js]: A `section` / `article` with a nested `ul` .

File: components/BattleScreen/PlayerState/playerstate.module.css [hunk 9]
```diff
@@ -0,0 +1,95 @@
+/* @font-face {
+	font-family: "pokemon-fire-red";
+	src: url("/public/Font/PressStart2P-Regular.ttf");
+	font-weight: normal;
+	font-style: normal;
+} */
+
+@import url("https://fonts.googleapis.com/css2?family=Press+Start+2P&display=swap");
```
Comment (PReichetanz) [inline components/BattleScreen/PlayerState/playerstate.module.css]: Use `next/font` instead.

File: components/BattleScreen/PlayerState/playerstate.module.css [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+.player--pokemon-health-bar {
+	position: absolute;
+	height: 5px;
+	width: 80px;
+	left: 111px;
+	top: 21px;
+	z-index: 10;
+	background-color: #25b527;
+	/* background: rgb(42,106,37);
+	background: linear-gradient(180deg, rgba(42,106,37,1) 0%, rgba(37,183,40,1) 39%); */
```
Comment (PReichetanz) [inline components/BattleScreen/PlayerState/playerstate.module.css]: Please remove unused code.

File: components/BattleScreen/Pokemon/EnemyPokemon.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import styles from "./enemypokemon.module.css";
+
+export default function EnemyPokemon() {
+	return (
+		<div className={styles["enemy-pokemon"]}>
+			<img src="/sprites/opponent/hard/mewtwo.gif" alt="mewtwo" />
```
Comment (PReichetanz) [inline components/BattleScreen/Pokemon/EnemyPokemon.js]: I think `eslint` wants you to use the `next/image` here instead.

File: components/BattleScreen/Pokemon/PlayerPokemon.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import styles from "./playerpokemon.module.css";
+
+export default function PlayerPokemon() {
+  return (
+    <div className={styles["player-pokemon"]}>
+      <img src="/sprites/starter/charizard-back.gif" alt="glurak" />
```
Comment (PReichetanz) [inline components/BattleScreen/Pokemon/PlayerPokemon.js]: `next/image`

File: components/Heading/index.test.js [hunk 13]
```diff
@@ -2,7 +2,7 @@ import Heading from ".";
 import { render, screen } from "@testing-library/react";
 
 test("test", () => {
-  render(<Heading>🐠</Heading>);
-  const element = screen.getByText("🐠");
-  expect(element).toBeInTheDocument();
+	render(<Heading>🐠</Heading>);
+	const element = screen.getByText("🐠");
+	expect(element).toBeInTheDocument();
```
Comment (PReichetanz) [inline components/Heading/index.test.js:7]: You are not using this, so you should remove it.

File: package.json [hunk 14]
```diff
@@ -11,20 +11,23 @@
...
+    "autoprefixer": "^10.4.14",
     "eslint": "8.35.0",
     "eslint-config-next": "13.2.3",
     "next": "13.2.3",
+    "postcss": "^8.4.21",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "styled-components": "^5.3.8"
+    "styled-components": "^5.3.8",
+    "tailwindcss": "^3.2.7"
```
Comment (PReichetanz) [inline package.json]: I think we should speak about the way you're going to write your CSS.

File: pages/index.js [hunk 15]
```diff
@@ -1,9 +1,10 @@
...
 export default function Home() {
-	return (
-		<main>
-			<Heading>Pokemon Battler</Heading>
-		</main>
-	);
+  return (
+    <>
+      <BattleScreen />
+    </>
```
Comment (PReichetanz) [inline pages/index.js]: You don't need the fragments here.

File: styles.js [hunk 16]
```diff
@@ -10,4 +10,11 @@ export default createGlobalStyle`
     margin: 0;
     font-family: system-ui;
   }
+
+  html,
+body {
+  padding: 0;
+  margin: 0;
+  font-family: -apple-system, BlinkMacSystemFont, Segoe UI,
```
Comment (PReichetanz) [inline styles.js]: Why don't you use the `body` selector already used above?

**Review by PReichetanz — CHANGES_REQUESTED** [review]
Nice work! I left some comments on issues to discuss.

Please reconsider
- styled-components over BEM,
- semantic HTML elements instead of `div`s.

File: components/BattleScreen/EnemyState/enemystate.module.css (hunk 1 above)
Comment (dennisbuchwald) [reply components/BattleScreen/EnemyState/enemystate.module.css]: I have reworked and installed the font into the stylen.js.

File: components/BattleScreen/EnemyState/enemystate.module.css (hunk 2 above)
Comment (dennisbuchwald) [reply components/BattleScreen/EnemyState/enemystate.module.css]: have revised the complete css.

File: components/BattleScreen/EnemyState/enemystate.module.css (hunk 3 above)
Comment (dennisbuchwald) [reply components/BattleScreen/EnemyState/enemystate.module.css]: I have done everywhere except where I will need it directly for the next US.

File: components/BattleScreen/EnemyState/EnemyState.js (hunk 4 above)
Comment (dennisbuchwald) [reply components/BattleScreen/EnemyState/EnemyState.js]: removed pretty much all <div> and replaced it with semantic HTML.

File: components/BattleScreen/EnemyState/EnemyState.js (hunk 5 above)
Comment (dennisbuchwald) [reply components/BattleScreen/EnemyState/EnemyState.js]: also completely reworked. unfortunately it didn't work like that. was replaced by a useState that will specify the prob to manipulate the display of the healthbar.

Regarding the list I'm still thinking, I decide as soon as the functionality comes to it.

File: components/BattleScreen/Pokemon/PlayerPokemon.js (hunk 12 above)
Comment (dennisbuchwald) [reply components/BattleScreen/Pokemon/PlayerPokemon.js]: modified

File: components/Heading/index.test.js (hunk 13 above)
Comment (dennisbuchwald) [reply components/Heading/index.test.js:7]: now i'm using it temporarily as a header, but will pretty much have no place in the final version of the app.

File: package.json (hunk 14 above)
Comment (dennisbuchwald) [reply package.json]: yes I have experimented with libarys 😁😅

File: styles.js (hunk 16 above)
Comment (dennisbuchwald) [reply styles.js]: i had general comprehension problems in CSS, made up for it yesterday and completely reworked the app.

File: components/BattleScreen/EnemyState/EnemyState.js [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+	top: 5px;
+	right: 50px;
+	z-index: 10;
+	text-shadow: 1px 1px 0px #b7b7b7;
+	font-size: 20px;
+	margin: 0;
+	padding: 0;
+`;
+
+const EnemyPokemonHealthBarContainer = styled.progress`
```
Comment (PReichetanz) [inline components/BattleScreen/EnemyState/EnemyState.js:50]: I didn't know that this HTML element exists, thank you!

File: components/BattleScreen/EnemyState/EnemyState.js [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+	transition: all 0.4s linear;
+	border-radius: 2.5px;
+`;
+
+export default function EnemyState() {
+	const [currentHealth, setCurrentHealth] = useState(120);
+	const maxHealth = 120;
+	const healthPercent = Math.floor((currentHealth / maxHealth) * 100);
+
+	// Funktion für das updaten der Healthbar
```
Comment (PReichetanz) [inline components/BattleScreen/EnemyState/EnemyState.js]: I don't think that you need this comment because the function is perfectly named ;)

File: components/BattleScreen/Menu/Menu.js [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+	color: black;
+	left: 100px;
+	top: 20px;
+	text-transform: uppercase;
+`;
+
+export default function Menu() {
+	return (
+		<MenuContainer className="menu">
+			<div className="menu--overview-box">
```
Comment (PReichetanz) [inline components/BattleScreen/Menu/Menu.js:85]: Because you don't have any menu yet, this should not be part of this PR.

File: components/BattleScreen/Menu/menu.module.css [hunk 20]
```diff
@@ -0,0 +1,93 @@
+.menu {
```
Comment (PReichetanz) [inline components/BattleScreen/Menu/menu.module.css:1]: Can you safely delete this file now?

File: components/BattleScreen/PlayerState/PlayerState.js [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+	margin: 0;
+	padding: 0;
+`;
+
+const PokemonHealthBarContainer = styled.progress`
+	position: absolute;
+	height: 6px;
+	width: 89px;
+	left: 88px;
+	top: 31px;
```
Comment (PReichetanz) [inline components/BattleScreen/PlayerState/PlayerState.js:66]: I really hope that these fixed values won't create any trouble regarding responsiveness!

File: components/BattleScreen/PlayerState/PlayerState.js [hunk 22]
```diff
@@ -0,0 +1,130 @@
...
+	margin: 0;
+	padding: 0;
+`;
+
+export default function PlayerState() {
+	const [currentHealth, setCurrentHealth] = useState(120);
+	const maxHealth = 120;
+	const healthPercent = Math.floor((currentHealth / maxHealth) * 100);
+
+	// Funktion für das updaten der Healthbar
```
Comment (PReichetanz) [inline components/BattleScreen/PlayerState/PlayerState.js]:
```suggestion
```

File: components/BattleScreen/Pokemon/EnemyPokemon.js [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+	return (
+		<EnemyPokemonContainer>
+			<Image
+				src="/sprites/opponent/hard/mewtwo.gif"
+				alt="mewtwo"
+				layout="intrinsic"
+				width={240}
+				height={240}
+			/>
+			{/* <EnemyPokemonCaption>Mewtwo</EnemyPokemonCaption> */}
```
Comment (PReichetanz) [inline components/BattleScreen/Pokemon/EnemyPokemon.js:26]:
```suggestion
```

File: next.config.js [hunk 24]
```diff
@@ -1,18 +1,33 @@
...
+		styledComponents: true,
+	},
+	images: {
+		remotePatterns: [
+			// {
+			//   protocol: 'https',
+			//   hostname: 'images.unsplash.com'
+			// },
+		],
+	},
```
Comment (PReichetanz) [inline next.config.js]: If you don't need it, you can remove it completely.

**Review by PReichetanz — COMMENTED** [review]
Nice work! I've only found some minor issues, but they are open for discussion.

File: components/BattleScreen/EnemyState/EnemyState.js (hunk 17 above)
Comment (dennisbuchwald) [reply components/BattleScreen/EnemyState/EnemyState.js:50]: I did not know this morning yet, thanks internet! :-D It can be used well to display progress bars, for example, I have then converted to a HP display.

File: components/BattleScreen/EnemyState/EnemyState.js (hunk 18 above)
Comment (dennisbuchwald) [reply components/BattleScreen/EnemyState/EnemyState.js]: yes, you are right.

File: components/BattleScreen/Menu/Menu.js (hunk 19 above)
Comment (dennisbuchwald) [reply components/BattleScreen/Menu/Menu.js:85]: Yes, these are actually fragments from before as well, shouldn't have been in the PR, is for the later US.

File: components/BattleScreen/Menu/menu.module.css (hunk 20 above)
Comment (dennisbuchwald) [reply components/BattleScreen/Menu/menu.module.css:1]: Not yet, since I have not yet revised the menu, since it is also not actually used in this PR and also in this US. We deleted at the latest until the next US.

File: components/BattleScreen/PlayerState/PlayerState.js (hunk 21 above)
Comment (dennisbuchwald) [reply components/BattleScreen/PlayerState/PlayerState.js:66]: Yes, this is still a bigger issue. Not sure how to solve this perfectly.

File: next.config.js (hunk 24 above)
Comment (dennisbuchwald) [reply next.config.js]: 👍🏼

File: components/BattleScreen/Pokemon/EnemyPokemon.js (hunk 23 above)
Comment (dennisbuchwald) [reply components/BattleScreen/Pokemon/EnemyPokemon.js:26]: 👍🏼

File: components/BattleScreen/PlayerState/PlayerState.js (hunk 22 above)
Comment (dennisbuchwald) [reply components/BattleScreen/PlayerState/PlayerState.js]: 👍🏼
